internal/p2p/peermanager.go

```diff
@@ -799,15 +802,27 @@ func (m *PeerManager) Errored(peerID types.NodeID, err error) {
 	m.evictWaker.Wake()
 }
 
+func (m *PeerManager) Inactivate(peerID types.NodeID) error {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	peer := m.store.peers[peerID]
+	peer.Inactive = true
+
+	return m.store.Set(*peer)
+}
+
 // Advertise returns a list of peer addresses to advertise to a peer.
 //
-// FIXME: This is fairly naïve and only returns the addresses of the
-// highest-ranked peers.
+// It finds twice as many peers as the limit specifies, shuffles this
+// list, and then returns the limit. The goal of this is to gossip the
+// "best" peers but also ensure that more fresh peers also have a
+// chance.
```

Comment:
we always gossip the top 100 peers, no matter what. I think you're right that there's a starvation possibility, and we've started gossiping 100 of the top 200 peers, but gossiping all nodes in the peer store eventually is going to be hard. 

We could always gossip the ones that dial in? I think that adding a default score will eventually help. If you're one-degree away from a node that's actually connected then you'll definitely get it in your peer store.

Reply:
This seems like a reasonable place to start and has the advantage of being easy to add on top of what we're already doing. 

The other possibility we discussed was to iterate the whole peer store and randomly pick or skip each peer till we have 100. That allows a low probability of eventually getting _all_ peers, but still strongly emphasizes the ones near the head.

I don't have a strong feeling that we have to do that, though I think we already pull the entire peer store anyway so it might not be too much worse than the easy case. YMMV.

Reply:
Yeah this is the approach I was advocating in https://github.com/tendermint/tendermint/issues/8735

Reply:
I'm not committed to any particular solution.

Reply:
Chiming in to say that I'd also prefer a solution that allows even low score peers to be gossiped _sometimes_. If we consistently gossip high-scored peers, they will become full and then other nodes will have a harder time learning about other available peers.

Reply:
Ok, sorry folks, I've eventually come around to Willaim's perspective and deleted the default score.